PureBasicIDE/ColorSchemes.pb

```diff
@@ -0,0 +1,324 @@
+      ; the PB standard scheme for screenreader support. The 'Accessibility'
+      ; scheme has a special option to always use these colors, so it is not needed here.
+      ;
+      If *ColorScheme\IsIDEDefault Or *ColorScheme\IsAccessibility Or EnableAccessibility Or (Colors(#COLOR_Selection)\PrefsValue = #ColorSchemeValue_UseSysColor)
+        Colors(#COLOR_Selection)\PrefsValue      = GetSysColor_(#COLOR_HIGHLIGHT)
+        Colors(#COLOR_SelectionFront)\PrefsValue = GetSysColor_(#COLOR_HIGHLIGHTTEXT)
+      EndIf
+    CompilerEndIf
+    
+    For i = 0 To #COLOR_Last
+      If (Colors(i)\PrefsValue >= 0)
+        UpdatePreferenceSyntaxColor(i, Colors(i)\PrefsValue)
+      Else
+        Colors(i)\PrefsValue = GuessColorSchemeColor(*ColorScheme, i)
+        UpdatePreferenceSyntaxColor(i, Colors(i)\PrefsValue)
+      EndIf
+    Next i
+    
+    DisableSelectionColorGadgets(*ColorScheme)
+    
+    If (PreferenceToolsPanelFrontColor < 0)
+      PreferenceToolsPanelFrontColor = GuessColorSchemeColor(*ColorScheme, #COLOR_ToolsPanelFrontColor)
+    EndIf
+    If (PreferenceToolsPanelBackColor < 0)
+      PreferenceToolsPanelBackColor = GuessColorSchemeColor(*ColorScheme, #COLOR_ToolsPanelBackColor)
+    EndIf
+    
+    If IsImage(#IMAGE_Preferences_ToolsPanelFrontColor)
+      UpdateImageColorGadget(#GADGET_Preferences_ToolsPanelFrontColor, #IMAGE_Preferences_ToolsPanelFrontColor, PreferenceToolsPanelFrontColor)
+    EndIf
+    If IsImage(#IMAGE_Preferences_ToolsPanelBackColor)
+      UpdateImageColorGadget(#GADGET_Preferences_ToolsPanelBackColor, #IMAGE_Preferences_ToolsPanelBackColor, PreferenceToolsPanelBackColor)
+    EndIf
+    
+  EndIf
+EndProcedure
+
+; Find and remove a known color scheme by its name
+Procedure RemoveColorSchemeIfExists(Name.s)
+  If (Name <> "")
+    ForEach ColorScheme()
+      If (ColorScheme()\Name = Name)
+        DeleteElement(ColorScheme())
+        Break
+      EndIf
+    Next
+  EndIf
+EndProcedure
+
+; Read the specified *ColorScheme from data section (for built-in schemes)
+Procedure.i ReadColorSchemeFromDataSection(*ColorScheme.ColorSchemeStruct)
+  If (*ColorScheme)
+    ; This assumes the NAME STRING data has already been read!
+    With *ColorScheme
+      \File = ""
+      Read.l \ColorValue[#COLOR_ToolsPanelFrontColor]
+      Read.l \ColorValue[#COLOR_ToolsPanelBackColor]
+      For i = 0 To #COLOR_Last
+        Read.l \ColorValue[i]
+      Next i
+    EndWith
+  EndIf
+  
+  ProcedureReturn (*ColorScheme)
+EndProcedure
+
+; Load the specified *ColorScheme from file on disk (for external schemes)
+Procedure.i LoadColorSchemeFromFile(*ColorScheme.ColorSchemeStruct, File.s)
+  Protected Result.i = #Null
+  
+  If (File)
+    ; Basic validation of color scheme file...
+    If (OpenPreferences(File))
+      Name.s = GetFilePart(File, #PB_FileSystem_NoExtension)
+      If (#True);(PreferenceGroup("Sections") And (ReadPreferenceLong("IncludeColors", 0) = 1))
```

Comment:
What is the purpose of this?

Reply:
Are you asking about the If (#True) ? I was doing some basic prefs validation there, but then I bypassed it to make the code more lenient in what it accepts as a color scheme file. The comment just shows what logic was formerly there. I can remove it completely or re-enable it.

Reply:
So either remove it or uncomment the original check. Leaving it as it is does not really make sense in the long term.